### PR TITLE
Replaced `pcap` dependency with our own `rawsocket` to make cross com…

### PR DESCRIPTION
**Pull request by aviramha**
…piling faster and easier.

Thank you for contributing to mirrord!

Please make sure you added a CHANGELOG entry & squashed your PR to one commit.

### CODE REVIEW

File: tests/src/sanity.rs [hunk 1]
```diff
@@ -606,7 +606,6 @@ mod tests {
         }
     }
 
-    #[ignore] // TODO: create integration test instead.
```
Comment (eyalb181) [inline tests/src/sanity.rs:609]: Why'd you remove it?

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,2 +1,4 @@
 [toolchain]
-channel = "nightly"
\ No newline at end of file
+channel = "nightly-2022-10-06"
```
Comment (eyalb181) [inline rust-toolchain.toml:2]: This doesn't seem intentional?

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: rust-toolchain.toml (hunk 2 above)
Comment (aviramha) [reply rust-toolchain.toml:2]: it is clippy is broken in newer versions and that's how I pin it globally..

File: tests/src/sanity.rs (hunk 1 above)
Comment (aviramha) [reply tests/src/sanity.rs:609]: want to see that it works since we don't have much coverage here, will re-add once it passes.

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**